npm/private/npm_package_store.bzl

```diff
@@ -195,17 +195,41 @@ def _npm_package_store_impl(ctx):
             virtual_store_directory = src_directory
         else:
             virtual_store_directory = ctx.actions.declare_directory(virtual_store_directory_path)
-            copy_directory_bin_action(
-                ctx,
-                src = src_directory,
-                dst = virtual_store_directory,
-                copy_directory_bin = ctx.toolchains["@aspect_bazel_lib//lib:copy_directory_toolchain_type"].copy_directory_info.bin,
-                # Hardlinking source files in external repositories as was done under the hood
-                # prior to https://github.com/aspect-build/rules_js/pull/1533 may lead to flaky build
-                # failures as reported in https://github.com/aspect-build/rules_js/issues/1412.
-                hardlink = ctx.attr.hardlink,
-                verbose = ctx.attr.verbose,
-            )
+            if src_directory.path.endswith(".tgz"):
+                # npm packages are always published with one top-level directory inside the tarball, tho the name is not predictable
```

Comment:
Is this convention? Are there docs for that somewhere?

Reply:
I only know this experimentally. I never found any docs that say why this is.

Reply:
For example, https://registry.npmjs.org/@gar/promisify/-/promisify-1.1.3.tgz has a top-level `package` directory while https://registry.yarnpkg.com/@types/js-yaml/-/js-yaml-4.0.5.tgz has a top-level `js-yaml` directory 🤷‍♂️

Reply:
Vast majority I've seen have a top-level `package`